synapse/http/request_metrics.py

```diff
@@ -117,21 +117,25 @@ def _get_in_flight_counts():
     Returns:
         dict[tuple[str, str], int]
     """
-    for rm in _in_flight_requests:
+    # Cast to a list to prevent it changing while the Prometheus
```

Comment:
s/cast/copy/ maybe, technically.

I assume this is an atomic operation from the point of view of python threading?

Reply:
Yes, this will be locked by the GIL.